R/filetracker_roclet.R

```diff
@@ -0,0 +1,22 @@
+#' Create roclet that tracks Rust source files changes.
+#' @returns A roclet.
+#' @export
+filetracker_roclet <- function() {
+  roxygen2::roclet("filetracker")
+}
+
+#' @method roclet_process roclet_filetracker
+#' @export
+roclet_process.roclet_filetracker <- function(x, blocks, env, base_path) {
+  list()
+}
+
+#' @method roclet_output roclet_filetracker
+#' @export
+roclet_output.roclet_filetracker <- function(x, results, base_path, ...) {
+  if (needs_compilation()) {
+    touch_makevars()
+    cli::cli_alert_warning("Run {.code devtools::document()} one more time to recompile Rust source.")
```

Comment:
Why run `document()` again and not `install()`? Wouldn't that be the normal sequence, or am I missing something?

Reply:
My understanding is that when we change `*.rs`, which is not tracked by the default build system, the roclet discovers something has been modified at the end of the pipeline. We `touch` `Makevars`, but we have no way to ask for recompilation, as this basically requires the whole pipeline to be rerun. 

In short, all roclets are executed **after** compilation has happenned and their execution is independent from each other. 
If we update wrappers in roclet, the wrapers and their tags will not be read by the roxygen2 roclet which emits documentation and `NAMESPACE` entries, and there is no way to bypass this without hijacking the build system. The same applies to the source code modification.